TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -214,15 +214,37 @@ public override bool Visit(Move move)
                 {
                     return true;
                 }
+
+                var senderIsAlphanumeric = false;
+                DataDefinition senderDataDefinition = null;
+                if (moveSimple.SendingVariable?.StorageArea?.Kind == StorageAreaKind.DataOrCondition
```

Comment:
Is testing of storage area kind required ? I mean if we can find it in the dictionary we can go on ? See also equivalent remark for line 233.

Reply:
The test on kind avoids having `FunctionCallResults`, which we do not want to check for now.
You are right when saying it is similar with the cast line 233.